tests/unit/test_generate.py

```diff
@@ -0,0 +1,215 @@
+import json
+import os
+import subprocess
+
+import flexmock
+import pytest
+
+
+from apigentools.commands.generate import GenerateCommand
+from apigentools.config import Config, LanguageConfig
+from apigentools.utils import run_command
+
+FIXTURE_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures")
+
+
+def test_get_missing_templates(tmpdir):
+    temp_dir = tmpdir.mkdir("missing_templates")
+    languages = ["python", "go", "ruby"]
+    args = flexmock(template_dir=temp_dir)
+    cmd = GenerateCommand({}, args)
+    missing = cmd.get_missing_templates(languages)
+    assert missing == languages
+
+
+def test_get_version_from_lang_oapi_config():
+    args = flexmock()
+    cmd = GenerateCommand({}, args)
+    with open(os.path.join(FIXTURE_DIR, "go_oapi_config.json")) as f:
+        oapi_config = json.loads(f.read())
+    result = cmd.get_version_from_lang_oapi_config(oapi_config)
+    assert result == "0.0.1"
+
+
+def test_get_image_name():
+    args = flexmock(generated_with_image="apigentools:latest")
+    cmd = GenerateCommand({}, args)
+    result = cmd.get_image_name()
+    assert result == "apigentools:latest"
```

Comment:
I may be misreading, but shouldn't this function actually return the git hash as the image name?

Reply:
The docstring on the function says this:         
 
>Assuming that this invocation of apigentools is running in an image and the specified image tag is `:latest`, this function will replace it with `:git-1234abc`. Otherwise it will return unmodified image name.

Which indicates to me that it will only return the hash when running in a container. But I might be wrong?